migration-engine/core/src/api/rpc.rs

```diff
@@ -71,7 +71,7 @@ impl RpcApi {
     pub async fn new(datamodel: &str) -> CoreResult<Self> {
         let mut rpc_api = Self {
             io_handler: IoHandler::default(),
-            executor: crate::migration_api(datamodel).await?,
+            executor: Arc::new(crate::migration_api(datamodel).await?),
```

Comment:
Can't we `Arc::new(*crate::migration_api(datamodel).await?),` instead?

Reply:
Oh, that could work, I'll try that.

Reply:
Fails with an error about not knowing the size of `dyn Trait` values statically.